node/node_genesis.go

```diff
@@ -34,28 +34,32 @@ const (
 	InitFreeFund = 100
 )
 
-// genesisInitializer is a shardchain.DBInitializer adapter.
+// GenesisInitializer is a shardchain.DBInitializer adapter.
+type GenesisInitializer interface {
+	InitChainDB(db ethdb.Database, shardID uint32) error
+}
+
 type genesisInitializer struct {
 	node *Node
 }
 
 // InitChainDB sets up a new genesis block in the database for the given shard.
 func (gi *genesisInitializer) InitChainDB(db ethdb.Database, shardID uint32) error {
-	shardState := core.CalculateInitShardState()
-	if shardID != 0 {
+	shardState := committee.IncorporatingStaking.Read(big.NewInt(0))
```

Comment:
What does IncorporatingStaking mean? Any better names？

Reply:
Like, 'a committee incorporating the use of staking'. Maybe something like like `committee.WithStakingEnabled` or if assume that this value will always be the singular instance of `MemberReader`, then could say `committee.Reader`